packages/account/src/providers/provider.ts

```diff
@@ -680,45 +691,42 @@ export default class Provider {
    * @param transactionRequest - The transaction request object.
    * @returns A promise.
    */
-  async estimateTxDependencies(transactionRequest: TransactionRequest): Promise<void> {
-    let missingOutputVariableCount = 0;
-    let missingOutputContractIdsCount = 0;
-    let tries = 0;
-
+  async estimateTxDependencies(transactionRequest: TransactionRequest): Promise<CallResult> {
     if (transactionRequest.type === TransactionType.Create) {
-      return;
+      return {
+        receipts: [],
+      };
     }
 
-    let txRequest = transactionRequest;
+    await this.estimatePredicates(transactionRequest);
```

Comment:
Because I moved the `shouldEstimatePredicates` check into `this.estimatePredicates`, I could remove the `if` block `if(txRequest.hasPredicateInput(){...}`